src/grammar.adoc

```diff
@@ -0,0 +1,208 @@
+[[sec:grammar]]
+== Domain Grammar
+
+:nrm: subs="+normal"
+:bnf: cols="10,^2,40",frame=none,grid=none,stripes=none,options="noheader"
+:sp2: {nbsp}{nbsp}{nbsp}{nbsp}
+
+This chapter presents Ion 1.1's _domain grammar_, by which we mean the grammar of the domain
+of values that drive Ion's encoding features.
+
+We use a BNF-like notation for describing various syntactic parts of a document,
+including Ion data structures. In such cases, the BNF should be interpreted loosely to
+accommodate Ion-isms like commas and unconstrained ordering of struct fields.
+
+All `()[]{}` below are literal tokens of Ion syntax.  Single-quoted `*'?'*` and `*'{asterisk}'*`
+denote literal Ion symbols, while unquoted |, ?, and {asterisk} are BNF notation.
+
+
+=== Documents
+
+TODO this section needs much work.
+
+[{bnf}]
+|===
+|document  |::=| _segment_*
+|segment   |::=| _ivm_?  _value_* _directive_?
+|directive |::=| _symtab-directive_ \| _encoding-directive_
+|===
+
+
+=== Encoding Directives
+
+[{bnf}]
+|===
+|encoding-directive |::=| `$ion_encoding::(` _retention_? _module-decl_*
+                                             _symtab_? _top-mactab_? `)`
+
+|retention |::=|  `(` `*retain*` _retainees_ `)`
+|retainees |::=|  `*'{asterisk}'*`  \|  _module-name_*
+
+|module-decl |::=| _dependency_  \|  _inline-module-def_
+|dependency  |::=| _load-decl_  \|  _use-decl_  \|  _import-decl_
+
+|use-decl    |::=|  `(` `*use*` _use-item_* `)`
+|use-item    |::=| _module-name_  \|  _load-decl_
+
+|symtab      |::=|  `(` `*symbol_table*` _symtab-item_* `)`
+|symtab-item |::=|  _module-name_  \|  `[` _text_* `]`
+
+|top-mactab  |::=|  `(` `*macro_table*` _module-name_* )
+
+|===
+
+
+==== Catalog Access
+
+[{bnf}]
+|===
+|load-decl       |::=|  `(` `*load*` _load-body_ `)`
+|import-decl     |::=|  `(` `*import*` _load-body_ `)`
```

Comment:
Yes.  

Note that I'm trying to avoid commentary in this chapter, more for reference.  (I tried to do a callout but that's not possible from inside a table)